Проект 13 [Сорокин, Иванов]

artemenm opened the pull request.

artemenm committed:
Add files via upload

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- пол пользователя
+CREATE TYPE SEX AS ENUM('male', 'female', 'other', 'prefer not to disclose'); 
```
Кажется опцию **prefer not to disclose** можно убрать и представлять ее в виде **NULL**

dbarashev replied:
Формально, NULL означает "неизвестность". То есть вы не знаете мнения пользователя о его гендере. Это не то же самое, что "предпочел не раскрывать".

BulatZuraev replied:
То есть можно знать мнение пользователя о его гендере, при том что он предпочел его не раскрывать?

ivanovmp replied:
Это и есть его мнение! Мы хотим знать сам факт, что пользователь предпочёл не раскрыть свой гендер.

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  tel_number TEXT NOT NULL UNIQUE, -- чтобы можно было авторизовываться по номеру телефона, и он был разным у разных пользователей
+  sex SEX,
+  date_of_birth DATE,
+  photo_file_path TEXT,
```
Мне кажется удобнее хранить изображения прямо в базе с помощью поля с типом **bytea**, чтобы улучшить консистентность данных.

marinskiy replied:
> Мне кажется удобнее хранить изображения прямо в базе с помощью поля с типом **bytea**, чтобы улучшить консистентность данных.

о классно я не знал такого

ivanovmp replied:
Как это улучшит консистентность данных? И почему на большинстве сайтов, что я знаю, изображения хранят в виде файлов и ссылок на них, а не прямо в базе данных (насколько я понимаю)? В ВКонтакте не консистентные данные?

BulatZuraev replied:
Я имею в виду, что если хранить в таблице путь, то надо поддерживать синхронизацию с системой хранения изображений и если в синхронизации допущена ошибка, то может возникнуть ситуация, когда в таблицу есть путь, а изображения по этому пути нет. Мне кажется хранить изображения прямо в таблице проще, к тому же если поле не _NULL_, то данные там точно есть. Я не утверждаю, что так нужно делать всегда, но думаю лучше начинать с простого способа.

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+CREATE TABLE CountryTable(
+  id SERIAL PRIMARY KEY,
+  name TEXT NOT NULL UNIQUE, -- чтобы не было двух одинаково названных стран
+  tax INT NOT NULL
```
Можно добавить проверку на неотрицательность поля **tax**.

ivanovmp replied:
Done f75f924

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  id SERIAL PRIMARY KEY,
+  host_id INT REFERENCES UserTable(id), -- жильё:хозяин N:1
+  latitude NUMERIC,
+  longitude NUMERIC,
```
Можно добавить ограничение на широту и долготу.

ivanovmp replied:
Done c77b4ef

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  room_count INT CHECK(room_count >= 0),
+  bed_count INT CHECK(bed_count >= 0),
+  max_people INT CHECK(max_people >= 0),
+  cleaning_cost NUMERIC CHECK(cleaning_cost >= 0)
```
Вроде бы задание подразумевает, что поля должны быть не пустыми.

ivanovmp replied:
Done b55f6e0

ivanovmp replied:
Done d150f56

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  guest_count INT CHECK(guest_count >= 0),
+  comment TEXT,
+  accepted BOOLEAN,
+  final_cost NUMERIC CHECK(final_cost >= 0),
```
Думаю все поля кроме **comment** должны быть не пустыми.

ivanovmp replied:
Done 183e335

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- В промежуток с date_range_start по date_range_end цена на жильё housing_id равнялась cost
+CREATE TABLE PriceTable(
+  housing_id INT NOT NULL REFERENCES HousingTable,
+  date_range_start DATE NOT NULL,
+  date_range_end DATE NOT NULL,
```
**date_range_end** не может быть раньше чем **date_range_start**.

ivanovmp replied:
Done 852136e

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- В промежуток с date_range_start по date_range_end цена на жильё housing_id равнялась cost
+CREATE TABLE PriceTable(
+  housing_id INT NOT NULL REFERENCES HousingTable,
+  date_range_start DATE NOT NULL,
+  date_range_end DATE NOT NULL,
```
**date_range_end** не может быть раньше чем **date_range_start**.

ivanovmp replied:
Done 852136e

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- информация об отзыве арендатора
+CREATE TABLE RenterReviewTable(
+  id SERIAL PRIMARY KEY,
+  renter_id INT REFERENCES UserTable(id), -- отзыв:арендатор N:1
+  housing_id INT REFERENCES HousingTable(id), -- отзыв:жильё N:1
```
Думаю стоит добавить уникальность тройки **renter_id, housing_id, date_of_departure**, чтобы пользователь мог оставить только один отзыв в один период проживания.

ivanovmp replied:
Я думаю, это не плохо, если в один период проживания можно будет оставить несколько отзывов, и в задании я не видел, что это плохо.

BulatZuraev replied:
Наверное это некритично, но может возникнуть странная ситуация, когда пользователь оставил негативный и положительный отзывы.

ivanovmp replied:
Понравился вкус водопроводной воды, не понравился цвет люстры.

dbarashev replied:
@ivanovmp  В ТЗ всё таки отзыв постоянно в единственном числе.

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- Детали заявки id о проживании пользователя user_id
+CREATE TABLE ApplicationTable(
+  id SERIAL PRIMARY KEY,
+  renter_id INT REFERENCES UserTable(id),
+  housing_id INT REFERENCES HousingTable(id),
```
Здесь указано поле **id**, а в таблице HousingUtilityTable нет, думаю стоит придерживаться какого-то варианта.

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- список развлечений
+CREATE TABLE EntertainmentTable(
+  id SERIAL PRIMARY KEY,
+  latitude NUMERIC,
+  longitude NUMERIC,
```
Почему тут нету NOT NULL?

ivanovmp replied:
Done 94363d2

BulatZuraev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  name TEXT NOT NULL,
+  date_range_start DATE NOT NULL,
+  date_range_end DATE NOT NULL,
+  genre_id INT REFERENCES EntertainmentGenreTable(id),
```
Почему тут нету NOT NULL?

ivanovmp replied:
Done 94363d2

BulatZuraev reviewed (commented):
Отличная работа, хотя можно было добавить больше ограничений.

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+CREATE TABLE CountryTable(
+  id SERIAL PRIMARY KEY,
+  name TEXT NOT NULL UNIQUE, -- чтобы не было двух одинаково названных стран
+  tax INT NOT NULL
```
Кажется, что лучше использовать тип ```NUMERIC```, так  как ```tax``` может быть не целым. Также можно добавить ограничение на неотрицательность этого поля.

ivanovmp replied:
Done 04ee961

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  room_count INT CHECK(room_count >= 0),
+  bed_count INT CHECK(bed_count >= 0),
+  max_people INT CHECK(max_people >= 0),
+  cleaning_cost NUMERIC CHECK(cleaning_cost >= 0)
```
> Пользователь, желающий сдать жильё в аренду, должен указать
большое количество разнообразных характеристик.

Так что поля, наверное, стоит сделать ```NOT NULL```.

ivanovmp replied:
Done 769035b

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  country_id INT REFERENCES CountryTable(id), -- жильё:страна N:1
+  address TEXT,
+  description TEXT,
+  room_count INT CHECK(room_count >= 0),
+  bed_count INT CHECK(bed_count >= 0),
+  max_people INT CHECK(max_people >= 0),
```
Может ли быть ситуация когда ```max_people``` или ```room_count``` равны 0?

ivanovmp replied:
Например, очень дешёвая квартира, где есть только кухня и туалет, которые не считаются за комнаты. Касательно числа людей — в принципе, может быть конференц-зал, не предназначенный для жилья, который, тем не менее, разрешено арендовать.

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- Список возможных удобств
+CREATE TABLE UtilityTable(
+  id SERIAL PRIMARY KEY,
+  name TEXT UNIQUE -- удобство задаётся своим описанием
```
Думаю, что ```name``` тоже можно сделать ```NOT NULL```.

ivanovmp replied:
Done 0f9c895

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  housing_id INT NOT NULL REFERENCES HousingTable,
+  date_range_start DATE NOT NULL,
+  date_range_end DATE NOT NULL,
+  cost INT CHECK(cost >= 0),
```
Поле ```cost``` также можно сделать ```NUMERIC```.

ivanovmp replied:
Done c3b4616

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- справочник жанров событий
+CREATE TABLE EntertainmentGenreTable(
+  id SERIAL PRIMARY KEY,
+  name TEXT UNIQUE -- название жанра идентифицирует этот жанр
```
Думаю, что ```name``` стоит сделать ```NOT NULL```.

ivanovmp replied:
Done 8ba89cd

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- В аренде housing_id есть удобство utility_id
+CREATE TABLE HousingUtilityTable(
+  housing_id INT NOT NULL REFERENCES HousingTable,
+  utility_id INT NOT NULL REFERENCES UtilityTable
```
Можно добавить ```UNIQUE (housing_id, utility_id)```

ivanovmp replied:
Done 1c4a529

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  surname TEXT NOT NULL,
+  email TEXT UNIQUE, -- чтобы можно было авторизовываться по адресу электронной почты, и он был разным у разных пользователей
+  tel_number TEXT NOT NULL UNIQUE, -- чтобы можно было авторизовываться по номеру телефона, и он был разным у разных пользователей
+  sex SEX,
```
Если в енуме есть опция ```prefer not to disclose```, то можно сделать поле ```sex``` ```NOT NULL```.

ivanovmp replied:
Done 16d92d5

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  accepted BOOLEAN,
+  final_cost NUMERIC CHECK(final_cost >= 0),
+  CHECK(arrival_date <= departure_date)
+);
```
Если я правильно понимаю логику работы с этой таблицей, то при формировании заявки поле ```accepted``` изначально ```NULL``` и только при подтверждении/отвержении заявки происходит заполнение булианом, так? Остальные поля (кроме ```comment```) при этом стоит сделать ```NOT NULL```.

ivanovmp replied:
А, ну да, логично.

ivanovmp replied:
Done fae829b

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+
+-- В промежуток с date_range_start по date_range_end цена на жильё housing_id равнялась cost
+CREATE TABLE PriceTable(
+  housing_id INT NOT NULL REFERENCES HousingTable,
+  date_range_start DATE NOT NULL,
+  date_range_end DATE NOT NULL,
```
Кажется, что для фиксированного дома на заданном интервале у нас не может быть несколько цен, так что стоит добавить ```UNIQUE(housing_id, date_range_start, date_range_end)```

dbarashev replied:
Там внизу определён такой ключ. Только неизменности на интервале это не гарантирует. Оставил начало таким же, конец поменял -- и вуаля, ключ счастлив, а пересечение интервалов тем не менее есть

artemenm replied:
@dbarashev В каком виде вы представляете проверку на отсутствие пересечений интервалов? Мне кажется, это не самая тривиальная логика для тех средств БД, с которыми мы знакомы.

dbarashev replied:
@ArtemSorokin возможно, дело в том, что не нужно усложнять себе жизнь, вводя интервалы?

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  location_rating RATING,
+  cleanness_rating RATING,
+  hospitality_rating RATING
+);
```
Возможно стоит сделать поля (кроме ```review_text```) ```NOT NULL```.

ivanovmp replied:
Done e01ac52

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  renter_id INT REFERENCES UserTable(id),
+  housing_id INT REFERENCES HousingTable(id),
+  review_text TEXT,
+  living_rating RATING
```
Можно сделать ```living_rating``` ```NOT NULL```

ivanovmp replied:
Done 903e20d

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  review_text TEXT,
+  living_rating RATING
+);
+-- других ключей нет
```
Возможно стоит пересмотреть архитектуру этих таблиц и привязать обе эти таблицы к согласованной заявке (из таблицы ```ApplicationTable```), тогда будет легче следить к какому именно событию относятся ревью.

artemenm replied:
Сделано в af95d59. Проблема шла с самой первой итерации.

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+CREATE TABLE EntertainmentTable(
+  id SERIAL PRIMARY KEY,
+  latitude NUMERIC,
+  longitude NUMERIC,
```
```latitude``` и ```longitude``` лучше сделать ```NOT NULL```.

ivanovmp replied:
Done 94363d2

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  date_range_end DATE NOT NULL,
+  genre_id INT REFERENCES EntertainmentGenreTable(id),
+  CHECK(date_range_start <= date_range_end)
+);
```
Можно ли предположить, что у нас не могут быть различные события с одинаковым временем, местом и названием? Если так, то можно добавить ```UNIQUE(latitude, longitude, name, date_range_start)```.

ivanovmp replied:
Думаю, что можно этого и не предполагать. Это довольно громоздкий ключ, особенно если учитывать, что сравнивать вещественные числа `latitude`, `longitude` на идеальное совпадение я вообще считаю не самой хорошей идеей.

ArtyomGrachev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- других ключей нет
+ 
+-- Детали заявки id о проживании пользователя user_id
+CREATE TABLE ApplicationTable(
+  id SERIAL PRIMARY KEY,
+  renter_id INT REFERENCES UserTable(id),
+  housing_id INT REFERENCES HousingTable(id),
+  arrival_date DATE,
+  departure_date DATE,
+  guest_count INT CHECK(guest_count >= 0),
+  comment TEXT,
+  accepted BOOLEAN,
+  final_cost NUMERIC CHECK(final_cost >= 0),
+  CHECK(arrival_date <= departure_date)
+);
```
Можно ли предположить, что один и тот же пользователь не может оставить несколько заявок на один и тот же дом на один и тот же интервал времени? Если так, то можно добавить ```UNIQUE(renter_id, housing_id, arrival_date, departure_date)```

ivanovmp replied:
А вдруг была создана заявка, она была отвергнута/отменена, а потом тот же пользователь подал новую заявку, практически такую же? Не вижу причины это запрещать.

IvanStrazov commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- пол пользователя
+CREATE TYPE SEX AS ENUM('male', 'female', 'other', 'prefer not to disclose'); 
+ 
+-- пользователь имеет данные данные
```
Не хватает описания таблиц по столбцам

ivanovmp replied:
Из задания:

> Для простых таблиц, описывающих очевидные свойства одного очевидно простого объекта, достаточно комментария вида "-- Строка представляет из себя объект Звездолёт с его свойствами".

Мы старались делать комментарии к каким-то нетривиальным связям. Разве надо расписывать, что `EntertainmentTable` — это развлечение `name` на широте `latitude` на долготе `longitude` жанра `genre_id`, проходящее в период от `date_range_start` до `date_range_end`? Это и так ясно из названий столбцов. Наверное, лучше исправить описания тех таблиц, в которых описывается скорее не объект, а связь? В каких таблицах такого рода недостаточные описания?

IvanStrazov commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+CREATE TABLE CountryTable(
+  id SERIAL PRIMARY KEY,
+  name TEXT NOT NULL UNIQUE, -- чтобы не было двух одинаково названных стран
+  tax INT NOT NULL
```
Можно сделать проверку комиссии в диапазоне от 0 до 100%

ivanovmp replied:
Комиссия измеряется не в процентах, а в талерах. И она необязательно меньше, чем основная стоимость проживания.

IvanStrazov commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+CREATE TABLE HousingTable(
+  id SERIAL PRIMARY KEY,
+  host_id INT REFERENCES UserTable(id), -- жильё:хозяин N:1
+  latitude NUMERIC,
```
Ограничения на координаты

ivanovmp replied:
Done f75f924

ivanovmp replied:
Done c77b4ef

IvanStrazov commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+);
+-- других ключей нет
+ 
+-- Список возможных удобств
```
Возможно, был смысл сделать удобства в виде типа данных. Но не принципиально

marinskiy replied:
> Возможно, был смысл сделать удобства в виде типа данных. Но не принципиально

Из условия: "Множество таких удобств довольно большое и может расширяться или
сужаться довольно часто". Так что с типом данных могут быть сложности, я бы оставил так)

ivanovmp replied:
Тоже думаю, что стоит оставить так.

IvanStrazov commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  genre_id INT REFERENCES EntertainmentGenreTable(id),
+  CHECK(date_range_start <= date_range_end)
+);
+-- ключей больше нет
```
В целом много мест без NOT NULL и явно не хватает описаний

ivanovmp replied:
Постарались сделать побольше `NOT NULL`ов. А где не хватает описаний?

marinskiy commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  id SERIAL PRIMARY KEY,
+  name TEXT NOT NULL,
+  surname TEXT NOT NULL,
+  email TEXT UNIQUE, -- чтобы можно было авторизовываться по адресу электронной почты, и он был разным у разных пользователей
```
на email достаточно легко регулярка пишется

dbarashev replied:
[Не совсем](https://stackoverflow.com/questions/201323/how-to-validate-an-email-address-using-a-regular-expression)

ivanovmp replied:
Не хочу загромождать код регулярной на email. Лучшая проверка на email — confirmation letter.

marinskiy commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  sex SEX,
+  date_of_birth DATE,
+  photo_file_path TEXT,
+  is_registered BOOLEAN NOT NULL);
```
Что то не могу в условии найти почему нам нужно is_registered

ivanovmp replied:
Странно, я, кажется, был уверен, что видел причины, почему поле `is_registered` должно быть. Но сейчас я этих причин не вижу.

ivanovmp replied:
Done 79bb121

marinskiy commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  latitude NUMERIC,
+  longitude NUMERIC,
+  country_id INT REFERENCES CountryTable(id), -- жильё:страна N:1
+  address TEXT,
```
Адрес  можно хранить не в виде текста, а как то более формализованно. Например в случае переименования какой то улицы мы можем столкнуться с достаточно нетривильной задачей.

dbarashev replied:
С другой стороны, у нас же не паспортный стол, а площадка с независимыми продавцами и покупателями. Пусть продавец сам меняет адрес, если переимееновалась улица.

marinskiy reviewed (commented):
Ребят хорошая работа вы молодцы) Критичных комментариев нет. Из некртиричного - во второй половине можно было бы добавить проверок + то что написал про адреса.

ArtyomGrachev reviewed (commented):
Хорошая работа. Оставил несколько комментариев по некоторым (по большей части ```NOT NULL``` ограничения) моментам.

dbarashev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+);
+-- других ключей нет
+
+-- В промежуток с date_range_start по date_range_end цена на жильё housing_id равнялась cost
```
Но требовались конкретно недели, а не произвольные промежутки времени.

artemenm replied:
Неделя — это неделя года, неделя с понедельника по воскресенье или просто любой промежуток длиной 7 дней?

ivanovmp replied:
Видимо, исправил

ivanovmp replied:
Done 074b25e

dbarashev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+  housing_id INT NOT NULL REFERENCES HousingTable,
+  date_range_start DATE NOT NULL,
+  date_range_end DATE NOT NULL,
+  cost INT CHECK(cost >= 0),
```
Было бы разумно таки выбрать один тип для денежных единиц. А то тут INT, там NUMERIC...

ivanovmp replied:
Done c3b4616

dbarashev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+-- информация об отзыве арендатора
+CREATE TABLE RenterReviewTable(
+  id SERIAL PRIMARY KEY,
+  renter_id INT REFERENCES UserTable(id), -- отзыв:арендатор N:1
+  housing_id INT REFERENCES HousingTable(id), -- отзыв:жильё N:1
+  review_text TEXT,
+  date_of_departure DATE,
```
renter_id, housing_id и date_of_departure как нибудь согласованы с другими такими же по смыслу атрибутами в других таблицах? Правда ли, например, что пользователь, написавший отзыв и поставивший рейтинги вообще мог никогда даже не подавать заявку?

artemenm replied:
Сделано в af95d59. Проблема шла с первой итерации.

dbarashev commented on project13/project13.sql:
```diff
@@ -0,0 +1,125 @@
+ 
+-- TODO: может можно Range(1,5);
+-- число звездочек в отзыве
+CREATE TYPE RATING AS ENUM('1', '2', '3', '4', '5');
```
Удобно будет среднее считать, я так думаю.

ivanovmp replied:
Нам приходит в голову только мысль, что можно сделать справочник:
```
CREATE TABLE rating(
	value INT PRIMARY KEY
);
```
И проверять, что рейтинг присутствует в этой таблице. Так будет лучше?

ivanovmp replied:
В эту таблицу можно поставить ограничение, что рейтинг должен быть от 1 до 5, а если рейтинговая система поменяется, то это ограничение будет находиться в одном месте, а не в пяти, и поменять его можно будет тоже в одном месте.

dbarashev replied:
@ivanovmp я бы просто поставил это ограничение там, где используется значение (в ревью тобишь), но если уж очень хочется заморочиться на красивости, можно сделать например [свой собственный домен](https://postgrespro.ru/docs/postgrespro/12/sql-createdomain)

dbarashev commented:
@ArtemSorokin @ivanovmp отметил эмоджей с глазками комментарии к недостаткам, причину и итерацию появления которых нужно выяснить

ivanovmp committed:
Add check of tax nonnegativity

ivanovmp committed:
Homogenize formatting

ivanovmp committed:
Restrict latitudes and longitudes

Actually, it's said in https://stackoverflow.com/questions/15965166/what-is-the-maximum-length-of-latitude-and-longitude that valid latitudes are from -85.05112878 to 85.05112878 degrees, but I decided to ignore such a nuance.

ivanovmp committed:
Add CHECK(start <= end)

ivanovmp committed:
Change type of tax

Tax actually should be NUMERIC, not INT

ivanovmp committed:
Make columns of HousingTable NOT NULL

ivanovmp committed:
Fix positions of NOT NULL in HousingTable

ivanovmp committed:
Make columns of EntertainmentTable NOT NULL

ivanovmp committed:
Remove is_registered from UserTable

Because this column is useless

ivanovmp committed:
Fix NOT NULL

ivanovmp committed:
Change type of cost

Cost actually should be NUMERIC, not INT

ivanovmp committed:
Make utility name NOT NULL

artemenm committed:
connect reviews to applications

ivanovmp committed:
Make all ratings not null

ivanovmp committed:
Make sex NOT NULL

ivanovmp committed:
Add UNIQUE (housing_id, utility_id)

ivanovmp committed:
Make NOT NULLs in ApplicationTable

ivanovmp committed:
Make ApplicationTable.accepted maybe null

ivanovmp committed:
Make reviews not null

ivanovmp committed:
Make EntertainmentGenreTable.name NOT NULL

ivanovmp committed:
Replace date range with week number

ivanovmp committed:
Syntax fix
